hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/TransferEventLoggingUtils.java

```diff
@@ -58,19 +61,12 @@ public static void logSuccessfulFungibleTransfer(
         requireNonNull(frame);
         requireNonNull(adjusts);
         requireNonNull(accountStore);
-        var senderId = AccountID.DEFAULT;
-        var receiverId = AccountID.DEFAULT;
-        long amount = 0L;
-        for (final var adjust : adjusts) {
-            amount = Math.abs(adjust.amount());
-            if (adjust.amount() > 0) {
-                receiverId = adjust.accountIDOrThrow();
-            } else {
-                senderId = adjust.accountIDOrThrow();
-            }
+        final var credit = adjusts.getFirst();
+        if (credit.amount() < 0) {
+            throw new IllegalArgumentException("Credit adjustment must appear first");
         }
-        frame.addLog(builderFor(tokenId, senderId, receiverId, accountStore)
-                .forDataItem(amount)
+        frame.addLog(builderFor(tokenId, adjusts.getLast().accountIDOrThrow(), credit.accountIDOrThrow(), accountStore)
```

Comment:
`adjusts.getLast()`. Do we know what will be the last adjust all the times ? 🤔 
Is this to make it behave like mono-service balance changes?

Reply:
This utility is only used to log ERC-style transfers (exactly two adjustments---one credit, one debit). And this method throws an `IllegalArgumentException` if the first adjustment is a debit. 

I.e., the input `List<AccountAmount>` always has the form `[(0.0.<receiver>, +X), (0.0.<sender>, -X)]`.